force-app/main/default/classes/selectors/tests/FundingRequestSelector_TEST.cls

```diff
@@ -0,0 +1,204 @@
+    }
+
+    @isTest
+    public static void getFundingRequestById() {
+        String fundingRequestId = [SELECT Id FROM Funding_Request__c LIMIT 1].Id;
+
+        test.startTest();
+
+        FundingRequestSelector fundingRequestSelector = new FundingRequestSelector();
+        List<Funding_Request__c> fundingRequests = fundingRequestSelector.getFundingRequestById(
+            fundingRequestId
+        );
+
+        test.stopTest();
+
+        system.assertNotEquals(
+            null,
+            fundingRequests,
+            'fundingRequests should have at least one funding request record'
+        );
+    }
+
+    @isTest
+    public static void callFundingRequestWithNoId() {
+        test.startTest();
+
+        FundingRequestSelector fundingRequestSelector = new FundingRequestSelector();
+        List<Funding_Request__c> fundingRequests = fundingRequestSelector.getFundingRequestById(
+            null
+        );
+
+        test.stopTest();
+
+        system.assertEquals(
+            new List<Funding_Request__c>(),
+            fundingRequests,
+            'fundingRequests should return an empty list'
+        );
+    }
+
+    @isTest
+    private static void callFundingRequest_MissingCrud() {
+        // Configure permissions.
+        FundingRequestSelector_TEST.minimalAccessPersona.load();
+        Test.startTest();
+
+        System.QueryException actualException;
+
+        System.runAs(FundingRequestSelector_TEST.minimalAccessPersona.getUser()) {
+            System.assertEquals(
+                false,
+                Schema.Funding_Request__c.SObjectType.getDescribe().isAccessible()
+            );
+
+            System.assertEquals(
+                false,
+                Schema.Contact.SObjectType.getDescribe().isAccessible()
+            );
+
+            System.assertEquals(
+                false,
+                Schema.Funding_Request__c.SObjectType.fields.Requested_Amount__c.getDescribe()
+                    .isAccessible()
+            );
+
+            System.assertEquals(
+                false,
+                Schema.Funding_Request__c.SObjectType.fields.Awarded_Amount__c.getDescribe()
+                    .isAccessible()
+            );
+
+            System.assertEquals(
+                false,
+                Schema.Funding_Request__c.SObjectType.fields.Total_Disbursed__c.getDescribe()
+                    .isAccessible()
+            );
+
+            System.assertEquals(
+                false,
+                Schema.Funding_Request__c.SObjectType.fields.Total_Remaining__c.getDescribe()
+                    .isAccessible()
+            );
+
+            System.assertEquals(
+                false,
+                Schema.Funding_Request__c.SObjectType.fields.Total_Remaining__c.getDescribe()
+                    .isAccessible()
+            );
+            // Question regarding lookups
+
+            // Run the test
+            String fundingRequestId = [SELECT Id FROM Funding_Request__c LIMIT 1].Id;
```

Comment:
Shouldn't we be retrieving the ID outside of the run-as block? I wouldn't expect this query to return a record as the min-access persona.

Reply:
That makes total sense let me move it up